### PR TITLE
title is set every time, but it could be empty

### PR DESCRIPTION
**Pull request by OskarStark**
Shall i change the other translation files too?

or does the fallback works automatically?

Oskar

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Can you provide a case where the title could be empty?

**Comment by OskarStark** [comment]
you could set this in your config:

```yml
- { position: left, type: sonata.timeline.block.timeline, settings: { title: '', context: GLOBAL, max_per_page: 25 }}
```

**Comment by OskarStark** [comment]
or am i wrong?
this is the code from the BlockService:
```php
    /**
     * {@inheritdoc}
     */
    public function buildEditForm(FormMapper $formMapper, BlockInterface $block)
    {
        $formMapper->add('settings', 'sonata_type_immutable_array', array(
            'keys' => array(
                array('title', 'text', array('required' => false)),
                array('max_per_page', 'integer', array('required' => true)),
            ),
        ));
    }
```

```php
    /**
     * {@inheritdoc}
     */
    public function setDefaultSettings(OptionsResolverInterface $resolver)
    {
        $resolver->setDefaults(array(
            'max_per_page'    => 10,
            'title'           => 'Latest Actions',
            'template'        => 'SonataTimelineBundle:Block:timeline.html.twig',
            'context'         => 'GLOBAL',
            'filter'          => true,
            'group_by_action' => true,
            'paginate'        => true,
        ));
    }
```

**Comment by Soullivaneuh** [comment]
I'll let @rande decide for that. :+1:

File: Resources/views/Block/timeline.html.twig [hunk 1]
```diff
@@ -12,7 +12,7 @@ file that was distributed with this source code.
 
 {% block block %}
     <div class="box box-primary">
-        {% if settings.title %}
+        {% if settings.title is not empty %}
```
Comment (Soullivaneuh) [inline Resources/views/Block/timeline.html.twig]: IMO, it shoulde be:

```twig
{% if settings.title and settings.title is not empty %}
```

**Comment by OskarStark** [comment]
i agree with you @Soullivaneuh, i fixxed it

**Comment by OskarStark** [comment]
ping @rande @Soullivaneuh anything else?

File: Resources/views/Block/timeline.html.twig [hunk 2]
```diff
@@ -12,7 +12,7 @@ file that was distributed with this source code.
 
 {% block block %}
     <div class="box box-primary">
-        {% if settings.title %}
+        {% if settings.title and settings.title is not empty %}
```
Comment (Soullivaneuh) [inline Resources/views/Block/timeline.html.twig]: After reflexion, I think `settings.title` and `settings.title is not empty` are partially the same, second is more complete.

Why not just keep the second condition?

Comment (OskarStark) [inline Resources/views/Block/timeline.html.twig]: i think you are right

Comment (OskarStark) [inline Resources/views/Block/timeline.html.twig]: done @Soullivaneuh

**Comment by Soullivaneuh** [comment]
Thank you @OskarStark.

**Pull request merged**